docs/sources/clients/promtail/logrotation.md

```diff
@@ -0,0 +1,132 @@
+# Promtail and Log rotation
+
+## Why log rotation matters?
+
+At any point in time, there can be three processes working on a log file as shown in the image below.
+![block_diagram](./logrotation-components.png)
+
+1. Appender - A writer that keep appending to a log file. This can be your application or some system daemons like syslog, docker log driver or kubelet etc.
+2. Tailer - A reader that read log lines as it is being appended e.g agents like Promtail.
+3. Log Rotater - A process that rotates the log either based on time (say scheduled every day) or size (say a log file reached its max size)
+
+NOTE: Here `fd` defines a file descriptor. Once a file is open for read or write, OS gives back unique file descriptor (an ingeter) per process and all the operations like read and write are done over that file descriptor. In other words, once file is opened successfully, file descriptor matters more than the file name.
+
+One of the critical component here is log rotater. Let's understand how it impacts other components like appender and tailer.
+
+In general, a log can be rotated in two ways.
```

Comment:
Good point. Here when we talk about "ways to logrotate", it's about "how" exactly log rotation works rather than "how often" it is rotated. May be need some different wording there. I will rephrase it. thanks :)